src/renderer/modals/configureServer/configureServer.tsx

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useState} from 'react';
+import ReactDOM from 'react-dom';
+
+import {TeamWithIndex} from 'types/config';
+import {ModalMessage} from 'types/modals';
+
+import {
+    MODAL_RESULT,
+    GET_MODAL_UNCLOSEABLE,
+    GET_DARK_MODE,
+    DARK_MODE_CHANGE,
+    RESIZE_MODAL,
+    MODAL_INFO,
+} from 'common/communication';
+import IntlProvider from 'renderer/intl_provider';
+import ConfigureServer from '../../components/ConfigureServer';
+
+import 'bootstrap/dist/css/bootstrap.min.css';
+
+const MOBILE_SCREEN_WIDTH = 1200;
+
+const onConnect = (data: TeamWithIndex) => {
+    window.postMessage({type: MODAL_RESULT, data}, window.location.href);
+};
+
+const ConfigureServerModalWrapper = () => {
+    const [darkMode, setDarkMode] = useState(false);
+    const [mobileView, setMobileView] = useState(false);
+    const [currentTeams, setCurrentTeams] = useState<TeamWithIndex[]>([]);
+
+    useEffect(() => {
+        setMobileView(window.innerWidth < MOBILE_SCREEN_WIDTH);
+        window.postMessage({type: GET_MODAL_UNCLOSEABLE}, window.location.href);
+        window.postMessage({type: GET_DARK_MODE}, window.location.href);
+        window.addEventListener('message', handleMessageEvent);
+
+        return () => {
+            window.removeEventListener('message', handleMessageEvent);
+        };
+    }, []);
+
+    const handleMessageEvent = (event: {data: ModalMessage<boolean | Electron.Rectangle | TeamWithIndex[]>}) => {
+        switch (event.data.type) {
+        case DARK_MODE_CHANGE:
+            setDarkMode(event.data.data as boolean);
+            break;
+        case RESIZE_MODAL: {
```

Comment:
Is there any reason we can't just do this with a `window.on('resize')` hook instead? I'd like to avoid any additional plumbing that doesn't require specific data from the main process that can't be found elsewhere.

Reply:
Ack'd. Updated